src/secrets-filesystem/src/lib.rs

```diff
@@ -32,7 +33,9 @@ impl SecretsController for FilesystemSecretsController {
             match op {
                 SecretOp::Ensure { id, contents } => {
                     // create will override an existing file
-                    let mut file = File::create(self.secrets_storage_path.join(format!("{}", id)))?;
+                    let file_path = self.secrets_storage_path.join(format!("{}", id));
+                    let mut file = File::create(file_path.clone())?;
+                    file.set_permissions(Permissions::from_mode(0o600))?;
```

Comment:
You can do the creation and mode setting all in one with `OpenOptions`: https://doc.rust-lang.org/stable/std/fs/struct.OpenOptions.html#method.mode